Lib/ssl.py

```diff
@@ -327,12 +327,20 @@ def _inet_paton(ipname):
     Supports IPv4 addresses on all platforms and IPv6 on platforms with IPv6
     support.
     """
-    # inet_aton() also accepts strings like '1'
-    if ipname.count('.') == 3:
-        try:
-            return _socket.inet_aton(ipname)
-        except OSError:
-            pass
+    # inet_aton() also accepts strings like '1', '127.1', or
+    # '127.0.0.1 whatever'.
+    try:
+        addr = _socket.inet_aton(ipname)
+    except OSError:
+        pass
+    else:
+        if _socket.inet_ntoa(addr) == ipname:
```

Comment:
Any reason not to `ipname.rstrip()` here for the comparison? Does whitespace matter?

Reply:
I just notice that you're explicitly testing trailing whitespace below.

Reply:
``rstrip()`` would deal with invalid input like ``127.0.0.1 whatever``, ``127.0.0.1\tinvalid``, or other embedded white spaces.

Reply:
> I just notice that you're explicitly testing trailing whitespace below.

The ``rstrip`` call in ``_ipaddress_match`` is necessary because OpenSSL includes a trailing new line for IPv6 addresses. The code removes trailing white space in SAN ``IP Address`` items of decoded certs, not of the user supplied ``hostname`` argument.